File: cpp/228. Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int start = 0;
        int end = 0;
        stringstream ss;
        vector<string> ret;
        if(nums.empty())
            return ret;
        while(start < nums.size()){
            if(end + 1 < nums.size() && nums[end + 1] == nums[end] + 1){
                end++;
            }
            else{
                string s, startStr, endStr;
                if(start == end){
                    ss << nums[start];
                    ss >> s;
                    ss.clear();
                    ss.str("");
                }
                else{
                    ss << nums[start];
                    ss >> startStr;
                    ss.clear();
                    ss.str("");
                    ss << nums[end];
                    ss >> endStr;
                    ss.clear();
                    ss.str("");
                    s = startStr + "->" + endStr;
                }
                ret.push_back(s);
                start = 1 + end++;
            }
        }
        return ret;
    }
};

File: java/Array/228. Summary Ranges.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
  public List<String> summaryRanges(int[] nums) {
      if(nums.length == 0) return new ArrayList<>();

      List<String> res = new ArrayList<>();
      int start, end;

      for(start = 0; start < nums.length;) {
          end = start;
          while(end < nums.length - 1 && nums[end + 1] == nums[end] + 1) {
              end++;
          }

          // Generate string
          if(end == start) {
              res.add(nums[start] + "");
          }
          else {
              res.add(nums[start] + "->" + nums[end]);
          }
          start = end + 1;
      }

      return res;
  }
}
